### ISSUE TITLE
Snapshot with postgres `SERIAL` primary keys cannot be applied

### ISSUE THREAD

**Issue by nxmad** [directus/directus#10358]
### Preflight Checklist

- [X] I have completed all [Troubleshooting Steps](https://docs.directus.io/getting-started/support/#troubleshooting-steps).
- [X] I'm on [the latest version of Directus](https://github.com/directus/directus/releases).
- [X] There's [no other issue](https://github.com/directus/directus/issues?q=is%3Aissue) that already describes my problem.

### Describe the Bug

I'm trying to move my schema from local server to production machine since Directus 9RCsomething. And I always get different errors. Now I've upgraded to Directus 9.2.1 and schema apply still not working.
Result of schema snapshot - https://gist.github.com/nxmad/b27b57fa373b134891ab85a409beb7ba

When applying the snapshot, the default value `nextval(''categories_id_seq''::regclass)` is treated as a string, which fails.

### To Reproduce

1. Install latest Directus with postgres 13
2. run directus schema bootstrap
3. run directus schema apply

### Errors Shown

```
Invalid foreign key in field "group".
Error: Invalid foreign key in field "group".
    at foreignKeyViolation (/prod/node_modules/directus/dist/exceptions/database/dialects/postgres.js:101:12)
    at extractError (/prod/node_modules/directus/dist/exceptions/database/dialects/postgres.js:29:20)
    at translateDatabaseError (/prod/node_modules/directus/dist/exceptions/database/translate.js:50:56)
    at /prod/node_modules/directus/dist/services/items.js:95:68
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
```
    
If I nullify all groups:
```
error: create table "categories" ("id" integer not null default 'nextval(''categories_id_seq''::regclass)', "sort" integer null default null, "user_updated" uuid null default null, "date_updated" timestamptz null default null, "slug" varchar(255) not null default null, "game" integer null default null, "parent" integer null default null, "is_region" boolean not null default '0') - invalid input syntax for type integer: "nextval('categories_id_seq'::regclass)"       
    at Parser.parseErrorMessage (/prod/node_modules/pg-protocol/dist/parser.js:287:98)
    at Parser.handlePacket (/prod/node_modules/pg-protocol/dist/parser.js:126:29)
    at Parser.parse (/prod/node_modules/pg-protocol/dist/parser.js:39:38)
    at Socket.<anonymous> (/prod/node_modules/pg-protocol/dist/index.js:11:42)
    at Socket.emit (node:events:390:28)
    at addChunk (node:internal/streams/readable:315:12)
    at readableAddChunk (node:internal/streams/readable:289:9)
    at Socket.Readable.push (node:internal/streams/readable:228:10)
    at TCP.onStreamRead (node:internal/stream_base_commons:199:23)
```

### What version of Directus are you using?

9.2.1

### What version of Node.js are you using?

16.13.0

### What database are you using?

Postgres 13

### What browser are you using?

Chrome

### What operating system are you using?

Ubuntu 20.04

### How are you deploying Directus?

Docker

**Comment by viters** [comment]
I believe you should apply snapshots generated from the same version of Directus. At least that would be the least error-prone. In other words, if you have schema for version X, apply it to Directus v.X, then update that Directus instance to Y and generate snapshot of schema afterwards. You will have your snapshot, but ready for version Y.

**Comment by rijkvanzanten** [comment]
@viters is right. Update your 9RCsomething to v9.2.1 first, and then move the schema from 9.2.1 to another 9.2.1 👍🏻

**Comment by nxmad** [comment]
I guess there's little misunderstanding.
I've literally attached link of schema generated by Directus 9.2.1
So I'm trying to apply schema snapshot from local Directus 9.2.1 to remote Directus 9.2.1 deployment.
**Result of schema snapshot - https://gist.github.com/nxmad/b27b57fa373b134891ab85a409beb7ba**

**Comment by rijkvanzanten** [comment]
Ahh gotcha. That wasn't clear to me from your original issue 👍🏻 I thought you tried applying a snapshot from 9RCsomething to v9.2.1 🙂

**Comment by nxmad** [comment]
UPD. Looks like schema parser generates incorrect SQL.
This query is not working:
```
create table "categories" (
  "id" integer not null default 'nextval(''categories_id_seq''::regclass)',
  "sort" integer null default null,
  "user_updated" uuid null default null,
  "date_updated" timestamptz null default null,
  "slug" varchar(255) not null default null,
  "game" integer null default null,
  "parent" integer null default null,
  "is_region" boolean not null default '0'
)
```
`ERROR: invalid input syntax for type integer: "nextval('categories_id_seq'::regclass)" at character 62`
However, if I remove quotes around nextval invocation, everything is working fine as it supposed to do:
```
create sequence categories_id_seq start 1;

create table "categories" (
  "id" integer not null default nextval('categories_id_seq'::regclass),
  "sort" integer null default null,
  "user_updated" uuid null default null,
  "date_updated" timestamptz null default null,
  "slug" varchar(255) not null default null,
  "game" integer null default null,
  "parent" integer null default null,
  "is_region" boolean not null default '0'
);
```
I

**Comment by stefanfortuin** [comment]
I got the exact same problem.

Also trying to apply a snapshot generated with 9.2.1.

![image](https://user-images.githubusercontent.com/10186453/145203839-84db7cd3-affb-4cef-a0b2-6e492de44308.png)

Snapshot was generated with mysql:8 docker image
Trying to apply the snapshot on mariadb:10.5.13 docker image

Nodejs version 14.18

**Comment by Oreilles** [comment]
From the schema snapshot:
```yaml
fields:
  - collection: categories
    field: id
    type: integer
    schema:
      name: id
      table: categories
      data_type: integer
      default_value: nextval('categories_id_seq'::regclass)
      ...
      has_auto_increment: false
```

`has_auto_increment` is `false`, but it should be true.

What is the result of `SELECT pg_get_serial_sequence('categories', 'id') ?

**Comment by geryla** [comment]
Hi, I have similar issue.. when I want to apply my schema, I got error: 

Cannot read property 'COLUMN_TYPE' of undefined 

I did a lot of changes on my database after last schema applied.. But I need this functionality work properly.. it's very important for quick deployment process..

I'm sending my snapshot yaml. Thank you in advance!

My snapshot is here: https://codeshare.io/DZvE0Z

**Comment by geryla** [comment]
> Hi, I have similar issue.. when I want to apply my schema, I got error:
> 
> Cannot read property 'COLUMN_TYPE' of undefined
> 
> I did a lot of changes on my database after last schema applied.. But I need this functionality work properly.. it's very important for quick deployment process..
> 
> I'm sending my snapshot yaml. Thank you in advance!
> 
> My snapshot is here: https://codeshare.io/DZvE0Z

I renamed my local database to live database and got new error.. in column, what wasn't even touchted! :-(

alter table `events` modify `SEO` int null - ER_FK_COLUMN_CANNOT_CHANGE: Cannot change column 'SEO': used in a foreign key constraint 'events_seo_foreign'

**Comment by Oreilles** [comment]
@geryla this is unrelated, please first find out if there's an open issue looking like the one you have, then submit a new one with *clear* instructions and context if you haven't. The current issue you're answering to right now is a good example.

**Comment by nxmad** [comment]
> What is the result of `SELECT pg_get_serial_sequence('categories', 'id')`

It was NULL. I tried to fix all sequences with this function - https://gist.github.com/nxmad/973751c49acbc18019e8f5a2df247df0
Looks like this error is gone. But I still have `Invalid foreign key in field "group".`
And now it's trying to create directus_users collection from the scratch for some reason and I've got `Collections can't start with "directus_"`. I'm not sure if I need to create new issue for this or it's still related.
**Fixed schema snapshot - https://gist.github.com/nxmad/dde2fd69a3c792bd168e521c5308922e**
